### PR TITLE
feat: 15023 private public key operation

### LINKED ISSUE #15023: `pairings-signatures-library`: Implement the operations for creating the public/private key pairs
Following: https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/docs/proposals/TSS-Library/TSS-Library.md
Section: #Hedera Cryptography Pairings Signatures Library
Develop the possibility of creating private and public pairings ec key:\
SignatureSchema: class to select the parameters of the protocol (group for public keys and group for signing)
Private key is a random Scalar on the EC selected curve.
Public key is the generator point times the scalar of the group selected for public keys (we will need the class

### PR DESCRIPTION
**Pull request by mxtartaglia-sl**
**Description**:
Completing the private and public key creation operations

**Related issue(s)**:
Requieres: https://github.com/hashgraph/hedera-cryptography/pull/59

Fixes [#15023](https://github.com/hashgraph/hedera-services/issues/15023)

**Notes for reviewer**:
Still, binary files are provided until compiling support with Gradle for Rust is completed.

**Checklist**
- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/PairingPrivateKey.java [hunk 1]
```diff
@@ -44,6 +61,70 @@ public static PairingPrivateKey create(
...
+
+    /**
+     * Signs a message and returns the signature
+     *
+     * @return the signature of the message represented by {@code message}
+     */
+    @NonNull
+    public PairingSignature sign(final @NonNull byte[] message) {
+        Objects.requireNonNull(message, "message must not be null");
+        return new PairingSignature();
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/PairingPrivateKey.java]: this doesnt actually sign anything, should it throw an exception?

File: cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java [hunk 2]
```diff
@@ -70,6 +99,38 @@ public static SignatureSchema create(@NonNull final Curve curve, @NonNull final
...
+     * @return the SignatureSchema instance
+     */
+    @NonNull
+    public static SignatureSchema create(final byte idByte) {
+        byte curveId = BytePacker.unpackCurveType(idByte);
+        final Curve curve = PairingFriendlyCurves.allSupportedCurves().stream()
+                .filter(c -> c.getId() == curveId)
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Unknown curve id: " + idByte));
+        return new SignatureSchema(BytePacker.unpackGroupAssignment(idByte), curve);
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java:115]: I know its not part of this PR, but i think its easier to use `java.util.BitSet` inside `BytePacker`

File: cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Random;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.Test;
+
+class SignaturesLibraryTest {
+
+    @Test
+    void testFindAltBn128Provider() {
+        assertDoesNotThrow(() -> PairingFriendlyCurves.findInstance(KnownCurves.ALT_BN128));
```
Comment (lpetrovic05) [inline cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java:33]: ideally, asserts should also have a message

File: cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java (hunk 2 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java:115]: Taking a look at that class, but leaving the change for the last PR if that is ok with you.

File: cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java (hunk 3 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java:33]: Agreed, if it is ok with you, I will modify all tests in this project so they contain a message but in the latest pr, so I reduce the merge conflicts with draft prs already finished

File: cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/PairingPrivateKey.java (hunk 1 above)
Comment (mxtartaglia-sl) [reply cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/PairingPrivateKey.java]: Indeed, throwing an UnsupportedException.

File: cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java (hunk 2 above)
Comment (lpetrovic05) [reply cryptography/hedera-cryptography-pairings-signatures/src/main/java/com/hedera/cryptography/pairings/signatures/api/SignatureSchema.java:115]: sure

File: cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java (hunk 3 above)
Comment (lpetrovic05) [reply cryptography/hedera-cryptography-pairings-signatures/src/test/java/com/hedera/cryptography/pairings/signatures/api/SignaturesLibraryTest.java:33]: ok

**Review by lpetrovic05 — APPROVED** [review]
